frontend/public/src/containers/ProductDetailEnrollApp.js

```diff
@@ -55,7 +72,9 @@ export class ProductDetailEnrollApp extends React.Component<
     const { courseRuns } = this.props
     const { upgradeEnrollmentDialogVisibility } = this.state
     const product = run.products ? run.products[0] : null
-    const flexPriceDiscountAmount = product && product.product_flexible_price ? product.product_flexible_price.amount : 0
+    const flexDiscountAmount = product && product.product_flexible_price ? product.product_flexible_price.amount : 0
+    const flexDiscountType = product && product.product_flexible_price ? product.product_flexible_price.discount_type : null
+    const flexAdjustedCoursePrice = Number(calculateCoursePriceWithFlex(product.price, flexDiscountAmount, flexDiscountType)).toFixed(2)
```

Comment:
Nit: this gets repeated a couple times - can you refactor this into a utility function? Those are kept in `src/lib/utils.js`. Also - instead of `Number().toFixed()`, there's a utility function called `formatLocalePrice` that should be used here.